src/Stepper/Step.less

```diff
@@ -0,0 +1,205 @@
+@import "../less/fonts.less";
+@import (reference) "../less/index";
+
+@stepIconWidth: 1.75rem; // 28px
+@iconBorderWidth: @size_4xs; // 1px
+@iconFocusShadowThickness: @size_2xs; // 4px
+@titleFontSize: 1.25rem; // 20px
+@alertSize: 0.375rem; // 6px
+
+.Step {
+  .flexbox();
+  .alignItems(flex-start);
+  width: 100%;
+  .padding--y--2xs;
+}
+
+.Step--button.Button {
+  .padding--none;
+  text-align: left;
+  .text--line-height-3;
+}
+
+.Step--title {
+  display: block;
+  overflow: hidden;
+  color: @neutral_gray;
+  font-size: @titleFontSize;
+  .text--semi-bold;
+  .margin--top--2xs;
+  -ms-flex: 0 1 auto;
+}
+
+.Step--description {
+  color: @neutral_gray;
+  .text--medium;
+  .text--regular;
+}
+
+.Step--message--optional {
+  display: block;
+  overflow: hidden;
+  color: @neutral_gray;
+  .text--medium;
+  .text--italic;
+  .margin--bottom--xs;
+}
+
+.Step--message--warning {
+  display: block;
+  overflow: hidden;
+  color: @alert_red;
+  .text--medium;
+  .text--semi-bold;
+  .margin--bottom--xs;
+}
+
+.Step--icon {
+  display: inline-block;
+  .flexShrink(0);
+  .flexbox();
+  .alignItems(center);
+  .justifyContent(center);
+  .circle(@stepIconWidth);
+  background-color: @neutral_white;
+  background-position: center;
+  background-repeat: no-repeat;
+  background-size: contain;
+  border: @iconBorderWidth solid @neutral_gray;
+  .text--medium;
+  .text--regular;
+  .margin--right--s;
+  color: @neutral_gray;
+  box-shadow: 0 0 0 @iconFocusShadowThickness transparent;
+}
+
+.Step--success {
+  .Step--icon {
+    background-color: @neutral_dark_gray;
+    border: @iconBorderWidth solid @neutral_dark_gray;
+    color: transparent;
+  }
+}
+
+.Step--warning {
+  .Step--icon {
+    background-color: @alert_red;
+    border: @iconBorderWidth solid @alert_red;
+    color: transparent;
+  }
+}
+
+.Step--current {
+  .Step--icon {
+    background-color: @primary_blue;
+    border: @iconBorderWidth solid @primary_blue;
+    color: @neutral_white;
+  }
+
+  .Step--title, .Step--description, .Step--message--optional {
+    color: @neutral_dark_gray;
+  }
+}
+
+.Step--success:not(.Step--current) {
+  .Step--icon {
+    background-image: url("../../assets/images/check.svg");
```

Comment:
👍

Reply:
it's possible, but if you need the content to change on mouse interaction (from the mocks, it looks like we only need that for the `:active` state) then it's simpler to just treat the svg as an overlay and fade it out on `:active` (basically absolutely position it on top of the number text).

IMO, we should just push back on revealing the number on mouse down, since it doesn't add much, if anything at all. supposedly, if you click the step button, you're moving to that step anyway and you'll see the number, so i can't see much motivation for also revealing on mouse down.

Reply:
i think we might be miscommunicating - just to be sure, i was saying that we *shouldn't* bother with revealing the number on mouse down (which is the way is spec'd out in https://projects.invisionapp.com/share/6YPCS27X2JP#/screens/337902518_stepper_In_Dewey)

instead, just reveal the number when the page for that step is currently visible.

is that in line with what you were asking?

Reply:
yes, thanks